src/index.ts

```diff
@@ -1,44 +1,89 @@
-import loadWhenReady from './util/loadWhenDocumentReady'
-import { DefaultPageSettings } from './util/defaultPageSettings'
-export { PageSettingsBuilder } from './util/pageSettingsBuilder'
+/**
+ * Open Insights is a library for building customized RUM clients.
+ *
+ * @remarks
+ * This is the Open Insights core module. It defines business logic shared by
+ * all RUM clients. An Open Insights "provider" specifies the logic particular
+ * to a particular RUM service. Each of these is contained within its own
+ * package. A "tag owner" builds a RUM client in their own package by importing
+ * and utilizing features from the core module and one or more provider modules.
+ *
+ * @example
+ * ```typescript
+ * import { init, ClientSettingsBuilder } from 'open-insights'
+ * import { Provider, ProviderSettings } from 'open-insights-provider-foo'
+ *
+ * const settingsBuilder = new ClientSettingsBuilder()
+ * const fooSettings: ProviderSettings = {
+ *   setting1: 'some value',
+ *   setting2: 'some value,
+ * }
+ *
+ * settingsBuilder.addProvider(new Provider(fooSettings))
+ *
+ * // Execute a RUM session
+ * init(settingsBuilder.value)
+ *     .then(result => {
+ *         // `result` contains the results from the RUM session after
+ *         // completion
+ *     })
+ * ```
+ *
+ * @packageDocumentation
+ */
+import whenReady from "./util/loadWhenDocumentReady"
+import { ClientSettings, SessionResult } from "./@types"
 
-export function init(settings: PageSettings): void {
-    loadWhenReady(() => {
-        console.log('Document loaded and ready')
-        settings = settings || DefaultPageSettings
+/**
+ * Called by tag owner code to initialize a RUM session, either immediately or
+ * after some delay.
+ *
+ * @remarks
+ * Waits for the page to load before processing.
+ *
+ * @param settings Specifies settings affecting client behavior. These are
+ * determined by the tag owner at runtime, so may be used to specify page-level
+ * overrides to general defaults.
+ */
+export default function init(settings: ClientSettings): Promise<SessionResult> {
+    return whenReady().then(() => {
         if (settings.preConfigStartDelay) {
-            setTimeout(startLater(settings), settings.preConfigStartDelay)
-        } else {
-            start(settings)
+            return startLater(settings)
         }
+        return start(settings)
     })
 }
 
-function startLater(settings: PageSettings): () => void {
-    return function () {
-        start(settings)
-    }
+/**
+ * Called internally if a non-zero {@link ClientSettings.preConfigStartDelay}
+ * setting has been specified. Calls {@link start} after the delay.
+ *
+ * @param settings The settings object passed to {@link init}.
+ */
+function startLater(settings: ClientSettings): Promise<SessionResult> {
```

Comment:
While I know that the value is contained within the settings object, it feels weird to me that this function signature doesn't explicitly contain the delay value. Maybe:

```suggestion
function startLater(delay: number, settings: ClientSettings): Promise<SessionResult> {
```